test/gtest/uct/uct_test.cc

```diff
@@ -655,28 +697,56 @@ uct_test::mapped_buffer::mapped_buffer(size_t size, uint64_t seed,
 }
 
 uct_test::mapped_buffer::~mapped_buffer() {
-    m_entity.mem_free(&m_mem, m_rkey);
+    m_entity.mem_free(&m_mem, m_rkey, m_mem.mem_type);
 }
 
 void uct_test::mapped_buffer::pattern_fill(uint64_t seed) {
-    pattern_fill(m_buf, (char*)m_end - (char*)m_buf, seed);
+    pattern_fill(m_buf, (char*)m_end - (char*)m_buf, seed, m_mem.mem_type);
 }
 
-void uct_test::mapped_buffer::pattern_fill(void *buffer, size_t length, uint64_t seed)
+void uct_test::mapped_buffer::pattern_fill(void *buffer, size_t length, uint64_t seed,
+                                           uct_memory_type_t mem_type)
 {
     uint64_t *ptr = (uint64_t*)buffer;
     char *end = (char *)buffer + length;
 
-    while ((char*)(ptr + 1) <= end) {
-        *ptr = seed;
-        seed = pat(seed);
-        ++ptr;
+    if (mem_type == UCT_MD_MEM_TYPE_HOST) {
+        while ((char*)(ptr + 1) <= end) {
+            *ptr = seed;
+            seed = pat(seed);
+            ++ptr;
+        }
+        memcpy(ptr, &seed, end - (char*)ptr);
+    } else if (mem_type == UCT_MD_MEM_TYPE_CUDA) {
+#if HAVE_CUDA
+        void *temp;
```

Comment:
may be it is a good idea to move cuda handling code to separate functions.

Reply:
moved cuda pattern fill/check to separate funcs